firestore/integration_test.go

```diff
@@ -1037,46 +1046,66 @@ func TestIntegration_QueryDocuments(t *testing.T) {
 	}
 	q := coll.Select("q")
 	for i, test := range []struct {
-		q       Query
-		want    []map[string]interface{}
-		orderBy bool // Some query types do not allow ordering.
+		desc       string
+		q          Query
+		want       []map[string]interface{}
+		orderBy    bool // Some query types do not allow ordering.
+		orderByDir Direction
 	}{
-		{q, wants, true},
-		{q.Where("q", ">", 1), wants[2:], true},
-		{q.Where("q", "<", 1), wants[:1], true},
-		{q.Where("q", "==", 1), wants[1:2], false},
-		{q.Where("q", "!=", 0), wants[1:], true},
-		{q.Where("q", ">=", 1), wants[1:], true},
-		{q.Where("q", "<=", 1), wants[:2], true},
-		{q.Where("q", "in", []int{0}), wants[:1], false},
-		{q.Where("q", "not-in", []int{0, 1}), wants[2:], true},
-		{q.WherePath([]string{"q"}, ">", 1), wants[2:], true},
-		{q.Offset(1).Limit(1), wants[1:2], true},
-		{q.StartAt(1), wants[1:], true},
-		{q.StartAfter(1), wants[2:], true},
-		{q.EndAt(1), wants[:2], true},
-		{q.EndBefore(1), wants[:1], true},
-		{q.LimitToLast(2), wants[1:], true},
-		{q.EndBefore(2).LimitToLast(2), wants[:2], true},
-		{q.StartAt(1).EndBefore(2).LimitToLast(3), wants[1:2], true},
+		{"Without filters", q, wants, true, 0},
+		{"> filter", q.Where("q", ">", 1), wants[2:], true, Asc},
+		{"< filter", q.Where("q", "<", 1), wants[:1], true, Asc},
+		{"== filter", q.Where("q", "==", 1), wants[1:2], false, 0},
+		{"!= filter", q.Where("q", "!=", 0), wants[1:], true, Asc},
+		{">= filter", q.Where("q", ">=", 1), wants[1:], true, Asc},
+		{"<= filter", q.Where("q", "<=", 1), wants[:2], true, Asc},
+		{"in filter", q.Where("q", "in", []int{0}), wants[:1], false, 0},
+		{"not-in filter", q.Where("q", "not-in", []int{0, 1}), wants[2:], true, Asc},
+		{"WherePath", q.WherePath([]string{"q"}, ">", 1), wants[2:], true, Asc},
+		{"Offset with Limit", q.Offset(1).Limit(1), wants[1:2], true, Asc},
+		{"StartAt", q.StartAt(1), wants[1:], true, Asc},
+		{"StartAfter", q.StartAfter(1), wants[2:], true, Asc},
+		{"EndAt", q.EndAt(1), wants[:2], true, Asc},
+		{"EndBefore", q.EndBefore(1), wants[:1], true, Asc},
```

Comment:
Can we add a test case here that does q.startAfter(x).endBefore(y) where x > y without using limitToLast just to be sure?

Reply:
Added